pkg/manager/manager_test.go

```diff
@@ -430,72 +512,74 @@ func TestManager_GetAllConfig(t *testing.T) {
-	mgrTest, _, _, _, _, _ := setUp(t)
+	mgrTest, _, _, _ := setUp(t)
 	assert.Equal(t, mgrTest, GetManager())
 	GetManager().Close()
 	assert.Equal(t, mgrTest, GetManager())
 }
 
 func TestManager_ComputeRollbackDelete(t *testing.T) {
-	mgrTest, _, _, _, _, _ := setUp(t)
+	mgrTest, _, _, _ := setUp(t)
 
 	updates := make(devicechangetypes.TypedValueMap)
 	deletes := make([]string, 0)
-
 	updates[Test1Cont1ACont2ALeaf2B] = devicechangetypes.NewTypedValueFloat(ValueLeaf2B159)
-	err := mgrTest.ValidateNetworkConfig("Device1", "1.0.0", "", updates, deletes)
-	assert.NilError(t, err, "ValidateTargetConfig error")
-	_, _, err = mgrTest.SetNetworkConfig("Device1", "1.0.0", "", updates, deletes, "Testing rollback")
 
+	updatesForDevice1, deletesForDevice1, deviceInfo := makeDeviceChanges(Device1, updates, deletes)
+
+	err := mgrTest.ValidateNewNetworkConfig(Device1, DeviceVersion, DeviceType, updates, deletes)
+	assert.NilError(t, err, "ValidateTargetConfig error")
+	err = mgrTest.SetNewNetworkConfig(updatesForDevice1, deletesForDevice1, deviceInfo, "TestingRollback")
 	assert.NilError(t, err, "Can't create change", err)
 
 	updates[Test1Cont1ACont2ALeaf2B] = devicechangetypes.NewTypedValueFloat(ValueLeaf2B314)
 	updates[Test1Cont1ACont2ALeaf2D] = devicechangetypes.NewTypedValueFloat(ValueLeaf2D314)
 	deletes = append(deletes, Test1Cont1ACont2ALeaf2A)
 
-	err = mgrTest.ValidateNetworkConfig("Device1", "1.0.0", "", updates, deletes)
+	err = mgrTest.ValidateNewNetworkConfig(Device1, DeviceVersion, DeviceType, updates, deletes)
 	assert.NilError(t, err, "ValidateTargetConfig error")
-	changeID, configName, err := mgrTest.SetNetworkConfig("Device1", "1.0.0", "", updates, deletes, "Testing rollback 2")
 
+	err = mgrTest.SetNewNetworkConfig(updatesForDevice1, deletesForDevice1, deviceInfo, "TestingRollback2")
 	assert.NilError(t, err, "Can't create change")
 
-	_, changes, deletesRoll, errRoll := computeRollback(mgrTest, "Device1", *configName)
-	assert.NilError(t, errRoll, "Can't ExecuteRollback", errRoll)
-	config := mgrTest.ConfigStore.Store[*configName]
-	assert.Check(t, !bytes.Equal(config.Changes[len(config.Changes)-1], changeID), "Did not remove last change")
-	assert.Equal(t, changes[Test1Cont1ACont2ALeaf2B].Type, devicechangetypes.ValueType_FLOAT, "Wrong value to set after rollback")
-	assert.Equal(t, (*devicechangetypes.TypedFloat)(changes[Test1Cont1ACont2ALeaf2B]).Float32(), float32(ValueLeaf2B159), "Wrong value to set after rollback")
-
-	assert.Equal(t, changes[Test1Cont1ACont2ALeaf2A].Type, devicechangetypes.ValueType_FLOAT, "Wrong value to set after rollback")
-	assert.Equal(t, (*devicechangetypes.TypedFloat)(changes[Test1Cont1ACont2ALeaf2A]).Float32(), float32(ValueLeaf2B159), "Wrong value to set after rollback")
-
-	assert.Equal(t, deletesRoll[0], Test1Cont1ACont2ALeaf2D, "Path should be deleted")
+	// TODO - use new APIs to execute the roll back
+	//_, changes, deletesRoll, errRoll := computeRollback(mgrTest, Device1, *configName)
```

Comment:
I think you can use here computeNewRollback()

Reply:
I updated the test to use the new API